src/test/scala/com/mediative/eigenflow/publisher/MessagingSystemTest.scala

```diff
@@ -0,0 +1,16 @@
+package com.mediative.eigenflow.publisher
+
+import com.typesafe.config.{ ConfigValueFactory, ConfigObject, ConfigValue, ConfigFactory }
+import org.scalatest.FreeSpec
+
+class MessagingSystemTest extends FreeSpec {
+  "create" - {
+    "should create the appropriate messaging config" in {
+      val config = ConfigFactory.empty().withValue("eigenflow.messaging", ConfigValueFactory.fromAnyRef("com.mediative.eigenflow.publisher.PrintMessagingSystem"))
+
+      val messagingSystem = MessagingSystem.create(config)
+
+      assert(messagingSystem.isInstanceOf[PrintMessagingSystem])
+    }
+  }
```

Comment:
Can you add a case that asserts for failure when given an inappropriate messaging class name?